processing/src/main/java/io/druid/query/search/search/FragmentSearchQuerySpec.java

```diff
@@ -46,11 +67,21 @@ public FragmentSearchQuerySpec(
     return values;
   }
 
+  @JsonProperty
+  public boolean isCaseSensitive()
+  {
+    return caseSensitive;
+  }
+
   @Override
   public boolean accept(String dimVal)
   {
-    for (String value : values) {
-      if (dimVal == null || !dimVal.toLowerCase().contains(value.toLowerCase())) {
+    if (dimVal == null) {
+      return false;
+    }
+    final String input = caseSensitive ? dimVal : dimVal.toLowerCase();
+    for (String value : target) {
+      if (!input.contains(value)) {
```

Comment:
I think the docs are wrong here… they say fragment queries match if "any part of a dimension value contains any of the values specified" but the code matches only if all values are found.